src/t_zset.c

```diff
@@ -1985,7 +1985,7 @@ void zremrangeGenericCommand(client *c, zrange_type rangetype) {
          * The range is empty when start > end or start >= length. */
         if (start > end || start >= llen) {
             addReply(c,shared.czero);
-            goto cleanup;
+            return;
```

Comment:
I'm inclined to keep the clean up, so that even if in the future we extend the range checks to other types, we won't forget that cleanup is needed.